### PR TITLE
Moving appbuilder.xxx out of view files and into app.py

### PR DESCRIPTION
**Pull request by craig-rueda**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This is a follow-up to #8418 
* Moves `appbuilder.add_view` / `appbuilder.add_link` / etc. from view files into `app.py`
* Updates the docker_init.sh message a bit

### REVIEWERS
@dpgaspar @mistercrunch

### CODE REVIEW

File: superset/app.py [hunk 1]
```diff
@@ -114,10 +116,318 @@ def __call__(self, *args, **kwargs):
...
-    @staticmethod
-    def init_views() -> None:
-        # TODO - This should iterate over all views and register them with FAB...
-        from superset import views  # noqa pylint: disable=unused-variable
+    def init_views(self) -> None:
+        #
+        # We're doing local imports, as several of them import models which in turn try to import
+        # the global Flask app
+        #
+        from superset.connectors.druid.views import (
```
Comment (dpgaspar) [inline superset/app.py]: I would prefer if possible, to split this. For example on dashboards create a `DashboardManager` that would implement an init_views, then main `app.py` would invoke `init_views` for each manager.

So each package would be responsible for orderly adding what it implements.

Comment (craig-rueda) [reply superset/app.py]: Absolutely makes sense. I would like to do that as another step, however. This change will let us move in that direction as it removes some of the side effects that exist in the current setup. At some point, I'll pull relevant pieces out into their own "configuration" classes which will do just what you said - break things up and allow for a bit more composability. I'd also like to update the CLI WRT these changes.

Comment (mistercrunch) [reply superset/app.py]: There will be a fair amount of view shuffling as we break down and reorg modules. First goal is to move view init out of module scope, then find the right home for each view and init call. This is a good step in the right direction.

**Review by dpgaspar — APPROVED** [review]
LGTM

**Pull request merged**